src/containers/HomeTemplate/_components/courses/Courses.js

```diff
@@ -0,0 +1,213 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { actListCoursesApi } from "./modules/action";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faGraduationCap,
+  faEye,
+  faHeart,
+} from "@fortawesome/free-solid-svg-icons";
+import { NavLink } from "react-router-dom";
+import Swal from "sweetalert2";
+import AOS from "aos";
+import "aos/dist/aos.css";
+import { actAddToCart } from "containers/HomeTemplate/MyCart/module/actions";
+AOS.init();
+
+export default function Courses(props) {
+  const listCart = useSelector((state) => state.listCartReducer.listCart);
+  const [isHeartActive, setHeartActive] = useState(false);
+  const listCourses = useSelector(
+    (state) => state.listCoursesReducer.listCourses
+  );
+  // set state để hiển thị ra 3 item trước khi click button Show more
+  const [showMore, setShowMore] = useState({
+    listCourses: [...listCourses],
+    itemsToShow: 5,
+    expanded: false,
+  });
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(actListCoursesApi());
+  }, []);
+
+  const scrollUp = () => {
+    window.scroll({
+      top: 850,
+      left: 0,
+      behavior: "smooth",
+    });
+  };
+
+  const addToCart = () => {
+    localStorage.getItem("UserHome")
+      ? dispatch(actAddToCart(listCart))
+      : Swal.fire({
+          position: "center",
+          icon: "error",
+          html: `<h3 style="color:#f27474"><b>ERROR!</b></h3><b>Please LOG IN TO CONTINUE </b>`,
+          showConfirmButton: false,
+          timer: 1500,
+        });
+  };
+  const renderAddToCart = () => {
+    return listCart.findIndex((item) => {
+      return item.maKhoaHoc === listCourses.maKhoaHoc;
+    }) === -1 ? (
+      <button className="add-cart" onClick={() => {
+        addToCart()
+      }}>
+        Add To Cart
+      </button>
+    ) : (
+      <NavLink className="add-cart" to="/my-cart">
+        Go To Cart
+      </NavLink>
+    );
+  };
+  console.log(listCart);
+  // const handleAddToCart = () => {

```

Comment:
Unused + commented code like this should be removed otherwise it will become **zombie-code**